domain/storageprovisioning/service/filesystem.go

```diff
@@ -209,6 +210,23 @@ type FilesystemState interface {
 	SetFilesystemAttachmentProvisionedInfo(ctx context.Context, filesystemAttachmentUUID storageprovisioning.FilesystemAttachmentUUID, info storageprovisioning.FilesystemAttachmentProvisionedInfo) error
 }
 
+// CharmState defines the methods required to fetch the mount points for charm
+// containers.
+type CharmState interface {
+	// GetCharmUUIDForApplication returns the charm UUID for a given application.
+	GetCharmUUIDForApplication(
+		context.Context,
+		coreapplication.UUID,
+	) (corecharm.ID, error)
+	// GetContainerMountsForCharm returns the map of mount locations for workload
+	// containers. The map entry is keyed by the storage name.
+	// An empty map will be returned if there are no records.
+	GetContainerMountsForCharm(
```

Comment:
I would be happy if you wanted to consolidate these two now seeing the code up front. That is to say you can just have a func that goes 'GetContainermountsForApplication'. We have a strong uuid type to reference and on we can work our way across through RI.